app/javascript/shared/http.ts

```diff
@@ -22,18 +23,7 @@ const hooks: Hooks = {
   ],
 };
 
-const csrfMetaTag = document.querySelector('meta[name="csrf-token"]');
-const csrfToken = csrfMetaTag?.getAttribute('content');
-
-if (csrfToken) {
-  hooks['beforeRequest'] = [
-    (request) => {
-      request.headers.set('X-CSRF-Token', csrfToken);
-    },
-  ];
-}
-
-const kyApi = ky.extend({
+const kyApi = generalKyApi.extend({
```

Comment:
I'm drying up our `ky` usage, so that we don't need to repeat the CSRF header configuration (as we were doing before) and so that I only need to specify `retry: 0` in one place in this PR.